src/o365/spo/commands/contenttype/contenttype-remove.ts

```diff
@@ -0,0 +1,251 @@
+              method: 'POST',
+              headers: {
+                authorization: `Bearer ${siteAccessToken}`,
+                'X-HTTP-Method': 'DELETE',
+                'If-Match': '*',
+                'accept': 'application/json;odata=nometadata'
+              },
+              json: true
+            };
+
+            return request.post(requestOptions);
+          }
+          else {
+            return Promise.resolve({"odata.null":true});
+          }
+         
+      })
+      .then((res): void => {
+          //no response object expected
+          if (this.debug) {
+            cmd.log("deletion response:");
+            cmd.log(JSON.stringify(res));
+          }
+          
+            
+          if (res && res["odata.null"] === true) {
+            if (this.verbose) {
+              cmd.log(`Content type not found`);
+            }
+            cb(new CommandError(`Content type not found`));
+            return;
+          }
+          else {
+            
+            cmd.log("DONE");
+            
+          }
+          
+          cb();
+          
+        }, (err: any): void => {
+          this.handleRejectedODataJsonPromise(err, cmd, cb);
+        });
+
+      }
+
+      if (args.options.confirm) {
+        removeContentType();
+      }
+      else {
+        cmd.prompt({ type: 'confirm', name: 'continue', default: false, message: `Are you sure you want to remove the content type ${args.options.id||args.options.name}?`}, (result: { continue: boolean }): void => {
+          if (!result.continue) {
+            cb();
+          }
+          else {
+            removeContentType();
+          }
+        });
+      }
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --webUrl <webUrl>',
+        description: 'Absolute URL of the site where the content type exists'
+      },
+      {
+        option: '-i, --id [id]',
+        description: 'The ID of the content type to remove'
+      },
+      {
+        option: '-n, --name [name]',
+        description: 'The name of the content type to remove if ID is unknown'
+      },
+      {
+        option: '--confirm',
+        description: 'Don\'t prompt for confirming removal of the content type'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.webUrl) {
+        return 'Required parameter webUrl missing';
+      }
+
+      const isValidSharePointUrl: boolean | string = SpoCommand.isValidSharePointUrl(args.options.webUrl);
+      if (isValidSharePointUrl !== true) {
+        return isValidSharePointUrl;
+      }
+
+      if (!args.options.id && !args.options.name) {
+        return 'Either id or name parameter is required missing';
```

Comment:
This error message is confusing. Something along the lines of `You need to specify either the id or the name` would be clearer.